src/utils/recursiveResolve.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * wait recursively so that all promises are resolved
+ * need to go through all the nested objects and check if it is a promise
+ * if it is a promise, then await it
+ * @param object
+ * @returns
+ */
```

Comment:
I have no problem to use copilot to generate code, but please update the documentation for a human now :)